met-web/src/components/comments/admin/textListing/index.tsx

```diff
@@ -53,11 +66,23 @@ const CommentTextListing = () => {
                 sort_order,
                 search_text: searchFilter.value,
             };
+
             const response = await getSubmissionPage({
                 survey_id: Number(surveyId),
                 queryParams,
             });
-            setSubmissions(response.items);
+
+            const canViewAllComments =
+                UserService.hasAdminRole() ||
+                assignedEngagements.includes(survey.engagement_id) ||
+                UserService.hasRole('SuperUser');
+
+            const filterCondition = (submission: SurveySubmission) =>
```

Comment:
The filtering must happen in the backend, this will still load all comments into the browser.